flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -227,6 +241,12 @@ private static ClusterSpecification createClusterSpecification(CustomCommandLine
 		}
 	}
 
+	private Catalog createCatalog(String name, Map<String, String> catalogProperties, ClassLoader classLoader) {
+		final CatalogFactory factory =
+			TableFactoryService.find(CatalogFactory.class, catalogProperties, classLoader);
```

Comment:
Shall we rename the factory service, coz we are here creating catalogs rather than tables.

Reply:
I think it makes sense to rename the factory service and the factory interface itself to something more generic. I'll discuss with Timo, and it should be of its own PR.